MIKECoreCUtil/MIKECoreCUtil.cpp

```diff
@@ -0,0 +1,339 @@
+ *  @param[in] fp Specifies a pointer to the file.
+ *  @param[in] data Array to store data in, must be of size (num_timesteps * (num_items + 1))
+ *  @return
+ *  It returns a LONG that indicates possible errors that occurred during or before
+ *  the execution of the call. Possible return values are <b>F_NO_ERROR</b> ( = 0), indicating
+ *  successful execution or a nonzero value ( != 0) of type @ref FioErrors specifying the exact type
+ *  of error.
+ */
+MIKECORECUTIL_API int ReadDfs0DataDouble(LPHEAD pdfs, LPFILE fp, double* data)
+{
+  int rc;
+
+  // Get time axis information - should also work for non-time axis
+  int num_timesteps;
+  double to_sec_scale;
+  bool is_time_equidistant;
+  if ((rc = dfsGetTimeAxisInfo(pdfs, &num_timesteps, &to_sec_scale, &is_time_equidistant)))
+    return rc;
+
+  double double_delete = dfsGetDeleteValDouble(pdfs);
+  float float_delete = dfsGetDeleteValFloat(pdfs);
+
+  // Number of dynamic items
+  int num_items = dfsGetNoOfItems(pdfs);
+
+  // Store simple type for every item
+  SimpleType* itemSimpleTypes = new SimpleType[num_items];
+
+  /***********************************
+   * Dynamic item information
+   ***********************************/
+  LONG          item_type;                     // Item EUM type id
+  LPCTSTR       item_type_str;                 // Name of item type
+  LPCTSTR       item_name;                     // Name of item
+  LONG          item_unit;                     // Item EUM unit id
+  LPCTSTR       item_unit_str;                 // Item EUM unit string
+  SimpleType    item_datatype;                 // Simple type stored in item, usually float but can be double
+
+  int maxItemElmts = 1;
+  for (int i_item = 1; i_item <= num_items; i_item++)
+  {
+    LPITEM pitem = dfsItemD(pdfs, i_item);
+
+    // Name, quantity type and unit, and datatype
+    rc = dfsGetItemInfo(pitem, &item_type, &item_type_str, &item_name, &item_unit, &item_unit_str, &item_datatype);
+    if (rc) return rc;
+    itemSimpleTypes[i_item - 1] = item_datatype;
+
+    // Number of elements in item, find the max (in case they are not all dfs0 files, to not do any array-out-of-bounds operations)
+    int numElmts = dfsGetItemElements(pitem);
+    if (numElmts > maxItemElmts)
+      maxItemElmts = numElmts;
+  }
+
+  // Buffer for reading from dfs
+  float*  dfsdataf = new float[maxItemElmts];
+  double* dfsdatad = new double[maxItemElmts];
+
+  // Reset file pointer to first dynamic item-timestep.
+  dfsFindBlockDynamic(pdfs, fp);
+
+  int pos = 0;
+  double time;
+  for (int i = 0; i < num_timesteps; i++)
+  {
+    for (int j = 0; j < num_items; j++)
+    {
+      double value;
+      if (itemSimpleTypes[j] == UFS_FLOAT)
+      {
+        rc = dfsReadItemTimeStep(pdfs, fp, &time, dfsdataf);
+        float valuef = dfsdataf[0];
+        if (valuef != float_delete) 
+          value = valuef;
+        else
+          value = double_delete;
```

Comment:
Is this consistent with previous implementations? Using double_delete instead of float_delete will make some mikeio tests fail due to https://github.com/DHI/mikeio/blob/172b7dc0ca57971cf9f85acc2bd10ba690d1f782/mikeio/dfs0.py#L73 
But this can of course be adopted if necessary